src/embrakes/main.cpp

```diff
@@ -58,57 +58,108 @@ void Main::run()
+            m_brake->sendClamp();
+          }
+          if (!f_brake->checkClamped()) {
+            f_brake->sendClamp();
           }
           Thread::sleep(1000);
-          brake_1->checkHome();
+          m_brake->checkHome();
+          f_brake->checkHome();
+
+          m_brake->checkBrakingFailure();
+          f_brake->checkBrakingFailure();
         }
         break;
       case data::State::kCalibrating:
-        if (brake_1->checkClamped()) {
-          brake_1->sendRetract();
+        if (m_brake->checkClamped()) {
+          m_brake->sendRetract();
+        }
+        if (f_brake->checkClamped()) {
+          f_brake->sendRetract();
         }
-        if (!brake_1->checkClamped()) {
+
+        if (!m_brake->checkClamped() && !f_brake->checkClamped()) {
           em_brakes_.module_status = ModuleStatus::kReady;
           data_.setEmergencyBrakesData(em_brakes_);
         }
+
         Thread::sleep(1000);
-        brake_1->checkHome();
+        m_brake->checkHome();
+        f_brake->checkHome();
+        m_brake->checkAccFailure();
+        f_brake->checkAccFailure();
         break;
       case data::State::kAccelerating:
-        brake_1->checkAccFailure();
+        m_brake->checkAccFailure();
+        f_brake->checkAccFailure();
         break;
       case data::State::kNominalBraking:
-        if (!brake_1->checkClamped()) {
-          brake_1->sendClamp();
+        if (!m_brake->checkClamped()) {
+          m_brake->sendClamp();
+        }
+        if (!f_brake->checkClamped()) {
+          f_brake->sendClamp();
         }
         Thread::sleep(1000);
-        brake_1->checkHome();
+        m_brake->checkHome();
+        f_brake->checkHome();
 
-        brake_1->checkBrakingFailure();
+        m_brake->checkBrakingFailure();
+        f_brake->checkBrakingFailure();
+        break;
+      case data::State::kEmergencyBraking:
+        if (!m_brake->checkClamped()) {
+          m_brake->sendClamp();
+        }
+        if (!f_brake->checkClamped()) {
+          f_brake->sendClamp();
+        }
+        Thread::sleep(1000);
+        m_brake->checkHome();
+        f_brake->checkHome();
+
+        m_brake->checkBrakingFailure();
+        f_brake->checkBrakingFailure();
         break;
       case data::State::kFinished:
-        if (tlm_data_.nominal_braking_command) {
-          if (brake_1->checkClamped()) {
-            brake_1->sendRetract();
+        if (!tlm_data_.nominal_braking_command) {
+          if (m_brake->checkClamped()) {
+            m_brake->sendRetract();
+          }
+          if (f_brake->checkClamped()) {
+            f_brake->sendRetract();
           }
           Thread::sleep(1000);
-          brake_1->checkHome();
-        } else if (!tlm_data_.nominal_braking_command) {
-          if (!brake_1->checkClamped()) {
-            brake_1->sendClamp();
+          m_brake->checkHome();
+          f_brake->checkHome();
+          m_brake->checkAccFailure();
+          m_brake->checkAccFailure();
```

Comment:
This should be `f_brake` instead of `m_brake`, right?

Reply:
yep!